lua/nvim-lsp-installer/servers/foam/init.lua

```diff
@@ -0,0 +1,17 @@
+local server = require("nvim-lsp-installer.server")
+local path = require("nvim-lsp-installer.path")
+local npm = require("nvim-lsp-installer.installers.npm")
+
+return function(name, root_dir)
+	return server.Server:new({
+		name = name,
+		root_dir = root_dir,
+		homepage = "https://github.com/FoamScience/foam-language-server",
+		languages = { "foam", "OpenFOAM" },
+		installer = npm.packages({ "foam-language-server" }),
+		default_options = {
+			--cmd_env = npm.env(root_dir),
+			cmd = { path.concat({ root_dir, "node_modules", "foam-language-server", "bin", "foam-ls" }), "--stdio" },
```

Comment:
Whenever it's possible, we don't override the `cmd` but instead rely on the defaults provided by lspconfig. All we have to do is ensure that the executable exists in path, which with npm packages is done like so:

```suggestion
			cmd_env = npm.env(root_dir),
```

Reply:
OK, this is not working for me;
The language server is downloaded to `~/.local/share/nvim/lsp_servers/foam` with its executable.
But without overriding the cmd, I get 0 clients attached to buffer. Any idea what's going on, I'm using configuration from (neovim/nvim-lspconfig#1643)
Even with globally installed package, if I don't override the cmd, the server is not getting fired up; a little confused I must say.
![2022-01-08_17-39](https://user-images.githubusercontent.com/34474472/148652256-16d6559a-86cf-4279-8e87-f1afe6ab6c40.png)

Reply:
I believe this happens if the server is unable to initialize

Reply:
OFC I forgot `--stdio`; thanks; all good now